File: backend/employee-service/src/main/java/com/eems/employee/employee/service/EmployeeService.java
package com.eems.employee.employee.service;

import java.util.List;

import org.springframework.data.domain.Pageable;

import com.eems.employee.common.response.PagedResponse;
import com.eems.employee.employee.dto.EmployeeRequestDTO;
import com.eems.employee.employee.dto.EmployeeResponseDTO;

public interface EmployeeService {

    EmployeeResponseDTO createEmployee(EmployeeRequestDTO requestDTO);

    EmployeeResponseDTO updateEmployee(Long id, EmployeeRequestDTO requestDTO);

    EmployeeResponseDTO getEmployeeById(Long id);

    

    List<EmployeeResponseDTO> getAllEmployees();

    void deleteEmployee(Long id);
    
    PagedResponse<EmployeeResponseDTO> getEmployees(
        Pageable pageable,
        String search);
}

File: frontend/src/services/employeeService.js
import api from "../api/axios";
import { API } from "../config/apiConfig";

const EmployeeService = {

    getAllEmployees() {
        return api.get(`${API.EMPLOYEE}/employees`);
    },

    getEmployeeById(id) {
        return api.get(`${API.EMPLOYEE}/employees/${id}`);
    },

    createEmployee(employee) {
        return api.post(`${API.EMPLOYEE}/employees`, employee);
    },

    updateEmployee(id, employee) {
        return api.put(`${API.EMPLOYEE}/employees/${id}`, employee);
    },

    deleteEmployee(id) {
        return api.delete(`${API.EMPLOYEE}/employees/${id}`);
    },

    getEmployees(
        page = 0,
        size = 10,
        sortBy = "id",
        sortDir = "asc",
        search = ""
    ) {
        return api.get(`${API.EMPLOYEE}/employees/page`, {
            params: {
                page,
                size,
                sortBy,
                sortDir,
                search
            }
        });
    }

};

export default EmployeeService;